src/main/scala/org/broadinstitute/dsde/firecloud/service/OntologyService.scala

```diff
@@ -4,40 +4,44 @@ import akka.actor.{Actor, Props}
 import akka.pattern._
 import com.typesafe.scalalogging.LazyLogging
 import org.broadinstitute.dsde.firecloud.Application
-import org.broadinstitute.dsde.firecloud.dataaccess.OntologyDAO
-import org.broadinstitute.dsde.firecloud.service.OntologyService.AutocompleteOntology
-import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
-import spray.json.DefaultJsonProtocol._
+import org.broadinstitute.dsde.firecloud.dataaccess.{OntologyDAO, ResearchPurposeSupport}
+import org.broadinstitute.dsde.firecloud.model.DataUse.{ResearchPurpose, ResearchPurposeRequest}
 import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.impOntologyTermResource
+import org.broadinstitute.dsde.firecloud.service.OntologyService.{AutocompleteOntology, ResearchPurposeQuery}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
 import spray.httpx.SprayJsonSupport
-import spray.json._
 import spray.json.DefaultJsonProtocol._
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object OntologyService {
   sealed trait OntologyServiceMessage
   case class AutocompleteOntology(term: String) extends OntologyServiceMessage
+  case class ResearchPurposeQuery(researchPurposeRequest: ResearchPurposeRequest) extends OntologyServiceMessage
 
```

Comment:
I'm not sure I would have put the ResearchPurposeQuery in OntologyService. is this because they are under the same /duos endpoint? If we want to have a separate service for the Data Use Aware search API, I think it would be separate from the Autocomplete service (autocomplete needs an ES backend but our 2 APIs don't)

Reply:
actually, i think this is making more sense to me now. i forgot that our research purpose request needed to query ontology for parents. i think if we did a separate deploy we actually would consider deploying our apis with the consent-ontology service